From c823689f379e8a3fdff5ac0a9af5343850a93a97 Mon Sep 17 00:00:00 2001
From: Quentin Glidic <sardemff7+git@sardemff7.net>
Date: Fri, 22 Apr 2016 21:16:02 +0200
Upstream: no, https://github.com/php/php-src/pull/1885
Subject: [PATCH] Correctly check for pkg-config

It is a tool, and needs the correct prefix for cross-compiling.
Also, if $PKG_CONFIG is set, either by the user or AC_PATh_TOOL, we can
consider it will be executable.
A bare "test -x" is not enough for package managers which set
$PKG_CONFIG to <prefix>-pkg-config (not absolute).

Signed-off-by: Quentin Glidic <sardemff7+git@sardemff7.net>
---
 acinclude.m4            | 4 ++--
 ext/odbc/config.m4      | 4 ++--
 ext/pdo_pgsql/config.m4 | 4 ++--
 ext/zip/config.m4       | 4 ++--
 sapi/fpm/config.m4      | 6 +++---
 5 files changed, 11 insertions(+), 11 deletions(-)

diff --git a/acinclude.m4 b/acinclude.m4
index 70b1a79..55414f4 100644
--- a/acinclude.m4
+++ b/acinclude.m4
@@ -2329,11 +2329,11 @@ AC_DEFUN([PHP_SETUP_OPENSSL],[
 
   dnl First try to find pkg-config
   if test -z "$PKG_CONFIG"; then
-    AC_PATH_PROG(PKG_CONFIG, pkg-config, no)
+    AC_PATH_TOOL(PKG_CONFIG, pkg-config)
   fi
 
   dnl If pkg-config is found try using it
-  if test "$PHP_OPENSSL_DIR" = "yes" && test -x "$PKG_CONFIG" && $PKG_CONFIG --exists openssl; then
+  if test "$PHP_OPENSSL_DIR" = "yes" && test -n "$PKG_CONFIG" && $PKG_CONFIG --exists openssl; then
     if $PKG_CONFIG --atleast-version=0.9.8 openssl; then
       found_openssl=yes
       OPENSSL_LIBS=`$PKG_CONFIG --libs openssl`
diff --git a/ext/odbc/config.m4 b/ext/odbc/config.m4
index 5aa7efa..26631ca 100644
--- a/ext/odbc/config.m4
+++ b/ext/odbc/config.m4
@@ -393,9 +393,9 @@ PHP_ARG_WITH(iodbc,,
   AC_MSG_CHECKING(for iODBC support)
   if test "$PHP_IODBC" != "no"; then
     if test -z "$PKG_CONFIG"; then
-      AC_PATH_PROG(PKG_CONFIG, pkg-config, no)
+      AC_PATH_TOOL(PKG_CONFIG, pkg-config)
     fi 
-    if test -x "$PKG_CONFIG" && $PKG_CONFIG --exists libiodbc ; then
+    if test -n "$PKG_CONFIG" && $PKG_CONFIG --exists libiodbc ; then
       PHP_ADD_LIBRARY_WITH_PATH(iodbc, $PHP_IODBC/$PHP_LIBDIR)
       ODBC_TYPE=iodbc
       ODBC_INCLUDE=`$PKG_CONFIG --cflags-only-I libiodbc`
diff --git a/ext/pdo_pgsql/config.m4 b/ext/pdo_pgsql/config.m4
index 1b6dd0e..96d340d 100644
--- a/ext/pdo_pgsql/config.m4
+++ b/ext/pdo_pgsql/config.m4
@@ -73,8 +73,8 @@ if test "$PHP_PDO_PGSQL" != "no"; then
   if test $? -eq 0 ; then
     AC_MSG_RESULT([yes])
     dnl First try to find pkg-config
-    AC_PATH_PROG(PKG_CONFIG, pkg-config, no)
-    if test -x "$PKG_CONFIG" && $PKG_CONFIG --exists openssl; then
+    AC_PATH_TOOL(PKG_CONFIG, pkg-config)
+    if test -n "$PKG_CONFIG" && $PKG_CONFIG --exists openssl; then
       PDO_PGSQL_CFLAGS=`$PKG_CONFIG openssl --cflags`
     fi
   else
diff --git a/ext/zip/config.m4 b/ext/zip/config.m4
index 81f47df..f68a0da 100644
--- a/ext/zip/config.m4
+++ b/ext/zip/config.m4
@@ -53,7 +53,7 @@ if test "$PHP_ZIP" != "no"; then
 
   if test "$PHP_LIBZIP" != "no"; then
 
-    AC_PATH_PROG(PKG_CONFIG, pkg-config, no)
+    AC_PATH_TOOL(PKG_CONFIG, pkg-config)
 
     dnl system libzip, depends on libzip
     AC_MSG_CHECKING(for libzip)
@@ -62,7 +62,7 @@ if test "$PHP_ZIP" != "no"; then
       LIBZIP_LIBDIR="$PHP_LIBZIP/$PHP_LIBDIR"
       AC_MSG_RESULT(from option: found in $PHP_LIBZIP)
 
-    elif test -x "$PKG_CONFIG" && $PKG_CONFIG --exists libzip; then
+    elif test -n "$PKG_CONFIG" && $PKG_CONFIG --exists libzip; then
       if $PKG_CONFIG libzip --atleast-version 0.11; then
         LIBZIP_CFLAGS=`$PKG_CONFIG libzip --cflags`
         LIBZIP_LIBDIR=`$PKG_CONFIG libzip --variable=libdir`
diff --git a/sapi/fpm/config.m4 b/sapi/fpm/config.m4
index e6a1a49..1cd7a3a 100644
--- a/sapi/fpm/config.m4
+++ b/sapi/fpm/config.m4
@@ -588,12 +588,12 @@ if test "$PHP_FPM" != "no"; then
 
   if test "$PHP_FPM_SYSTEMD" != "no" ; then
     if test -z "$PKG_CONFIG"; then
-      AC_PATH_PROG(PKG_CONFIG, pkg-config, no)
+      AC_PATH_TOOL(PKG_CONFIG, pkg-config)
     fi
     unset SYSTEMD_LIBS
     unset SYSTEMD_INCS
 
-    if test -x "$PKG_CONFIG" && $PKG_CONFIG --exists libsystemd; then
+    if test -n "$PKG_CONFIG" && $PKG_CONFIG --exists libsystemd; then
       dnl systemd version >= 209 provides libsystemd
       AC_MSG_CHECKING([for libsystemd])
       SYSTEMD_LIBS=`$PKG_CONFIG --libs libsystemd`
@@ -601,7 +601,7 @@ if test "$PHP_FPM" != "no"; then
       SYSTEMD_VERS=`$PKG_CONFIG --modversion libsystemd`
       AC_MSG_RESULT([version $SYSTEMD_VERS])
 
-    elif test -x "$PKG_CONFIG" && $PKG_CONFIG --exists libsystemd-daemon; then
+    elif test -n "$PKG_CONFIG" && $PKG_CONFIG --exists libsystemd-daemon; then
       dnl systemd version < 209 provides libsystemd-daemon
       AC_MSG_CHECKING([for libsystemd-daemon])
       SYSTEMD_LIBS=`$PKG_CONFIG --libs libsystemd-daemon`
-- 
2.8.1

